### PR TITLE
fix markdown nits

### PR DESCRIPTION
**Pull request by miguelraz**
Fixed a bunch of nits the VSCode MarkdownLint extension had.

This sped up my editing experience a bit and I think the team should default to using it tbh.

### CODE REVIEW

**Review by listochkin — APPROVED** [review]

File: training-slides/src/control-flow.md [hunk 1]
```diff
@@ -122,10 +122,43 @@ fn main() {
     }
 ```
 
+## Foreshadowing! Pattern Matching with bindings 👻
```
Comment (jonathanpallant) [inline training-slides/src/control-flow.md:125]: Please don't put changes to content in with a bunch of whitespace changes, as it makes it hard to review.

Comment (miguelraz) [reply training-slides/src/control-flow.md:125]: Oh, definitely unintentional, lemme revert that.

Comment (miguelraz) [reply training-slides/src/control-flow.md:125]: Nice catch @jonathanpallant ! Thanks for that.

Comment (miguelraz) [reply training-slides/src/control-flow.md:125]: Yup, I didn't rebase this properly . Arghhhh

Comment (miguelraz) [reply training-slides/src/control-flow.md]: rebased 👯

**Comment by miguelraz** [comment]
Andrei noted in chat:
* We can enable this in CI with https://github.com/marketplace/actions/markdownlint-cli2-action
* We shouldn't wreck our `git blame`
* We can enable per file ignores with [`Disable one or more rules by name: 
`](https://github.com/DavidAnson/markdownlint/blob/main/README.md#configuration)

**Comment by listochkin** [comment]
IMO Let's look at current violations first, agree on the set of rules we want to enforce, build an config file, and only *then* do these sweeping changes.

**Comment by miguelraz** [comment]
Conclusion:

Not worth the hassle for now.

**Pull request closed without merging**